lib/pages/timetable/service/google_calendar_services.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:acs_upb_mobile/resources/google_apis.dart';
+import 'package:flutter_web_browser/flutter_web_browser.dart';
+import 'package:googleapis/calendar/v3.dart' as g_cal;
+import 'package:googleapis_auth/auth_io.dart';
+import 'package:acs_upb_mobile/resources/utils.dart';
+
+extension GoogleCalendarServices on UniEventProvider {
+  g_cal.Event convertEvent(UniEvent uniEvent) {
+    final g_cal.Event googleCalendarEvent = g_cal.Event();
+
+    final g_cal.EventDateTime start = g_cal.EventDateTime();
+    final DateTime startDateTime = uniEvent.start.toDateTimeLocal();
+
+    start
+      ..timeZone = 'Europe/Bucharest'
+      // google calendar has different timezone formats
+      ..dateTime = startDateTime;
+
+    final Duration duration = uniEvent.duration.toDuration();
+
+    final g_cal.EventDateTime end = g_cal.EventDateTime();
+    final DateTime endDateTime = startDateTime.add(duration);
+    end
+      ..timeZone = 'Europe/Bucharest'
+      ..dateTime = endDateTime;
+
+    final ClassHeader classHeader = uniEvent.classHeader;
+
+    /*
+    final EventReminders eventReminders = EventReminders();
+    final EventReminder eventReminder = EventReminder()
+      ..minutes = 30
+      ..method = 'popup';
+    eventReminders
+      ..overrides = <EventReminder>[eventReminder]
+      ..useDefault = false;
+    */
+    // TODO(bogpie): Require user to input how many minutes before a notification from GCal (including the "no notification" option)
+    googleCalendarEvent
+      //..reminders = eventReminders
+      ..start = start
+      ..end = end
+      ..summary = classHeader.acronym
+      ..colorId = (uniEvent.type.googleCalendarColor.index).toString()
+/*    ..description =  eventInstance.title ?? eventInstance.mainEvent.type
+              .toLocalizedString(context)*/
+      // TODO(bogpie): Use a relevant description, like type of class + lecturer - future PR
+      // TODO(bogpie): "Closest" color (from a list) - GCal works with limited no. of colors - future PR
+      ..location = uniEvent.location;
+
+    if (uniEvent is RecurringUniEvent) {
+      final String rruleBasedOnCalendarString = uniEvent.rruleBasedOnCalendar
+          .toString()
+          .replaceAll(RegExp(r'T000000'), 'T000000Z');
+      // TODO(bogpie): What if another rrule doesn't start at midnight? Will keep this implementation until my related 'rrule' package issue will be resolved.
+      print(rruleBasedOnCalendarString);
+      googleCalendarEvent.recurrence = [rruleBasedOnCalendarString];
+    }
+
+    return googleCalendarEvent;
+  }
+
+  Future<void> prompt(String url) async {
+    await FlutterWebBrowser.openWebPage(url: url);
+  }
+
+  Future<void> insertGoogleEvents(
+      List<g_cal.Event> googleCalendarEvents) async {
+    // This is the function for getting the user's consent to export events to Google Calendar. See comments in original method and in GoogleApiHelper
```

Comment:
Too many words that don't convey much info ("this is the function" yes I can see this is a function, thanks), + comments should be above what they refer to. What's the "original method" anyway?

Maybe just say "This opens a browser window asking the user to authenticate and allow access to edit their calendar"? And while we're at it, wouldn't this method make more sense in GoogleApiHelper anyway? It's uncommon to refer to the documentation of another (seemingly) unrelated class, it's an indication that maybe this should be in that class.

Reply:
Or maybe the other way around - the stuff in GoogleApiHelper could just be private methods in GoogleCalendarServices.